routers/home.go

```diff
@@ -75,42 +76,15 @@ func RenderRepoSearch(ctx *context.Context, opts *RepoSearchOptions) {
 		page = 1
 	}
 
-	var (
-		repos   []*models.Repository
-		count   int64
-		err     error
-		orderBy models.SearchOrderBy
-	)
-
-	ctx.Data["SortType"] = ctx.Query("sort")
-	switch ctx.Query("sort") {
-	case "newest":
-		orderBy = models.SearchOrderByNewest
-	case "oldest":
-		orderBy = models.SearchOrderByOldest
-	case "recentupdate":
-		orderBy = models.SearchOrderByRecentUpdated
-	case "leastupdate":
-		orderBy = models.SearchOrderByLeastUpdated
-	case "reversealphabetically":
-		orderBy = models.SearchOrderByAlphabeticallyReverse
-	case "alphabetically":
-		orderBy = models.SearchOrderByAlphabetically
-	case "reversesize":
-		orderBy = models.SearchOrderBySizeReverse
-	case "size":
-		orderBy = models.SearchOrderBySize
-	default:
-		ctx.Data["SortType"] = "recentupdate"
-		orderBy = models.SearchOrderByRecentUpdated
-	}
+	orderByType := utils.ParseRepoOrderByType(ctx.Query("sort"))
+	ctx.Data["SortType"] = utils.ToQueryString(orderByType)
```

Comment:
Why not add `ToQueryString` as a method of `RepoOrderByType`?

Reply:
Because `RepoOrderByType` is defined in `models/`, and `ToQueryString` is defined in `modules/util/`

Reply:
put them in the same package?

Reply:
see my other comment (https://github.com/go-gitea/gitea/pull/3116#discussion_r155998565)

Reply:
@ethantkoenig I still think the `ParseRepoOrderByType` and `ToQueryString` should stay with strut `RepoOrderByType`. These two functions is also DB-related. One is a construct of `RepoOrderByType` and another is a printer of `RepoOrderByType`.
And these two functions cannot be used without `RepoOrderByType`, so why split them on two packages?

Reply:
__tl;dr__: I don't see how `ParseRepoOrderByType` and/or `ToQueryString` are DB related; they parse query strings to/from repo sorting options, but in no way involve SQL queries.

__Long version:__ Here are my thoughts: The `RepoOrderByType` values each represent some way of sorting repositories (e.g. `RepoOrderByAlphabetically` represents the abstract notion of sorting repos by name). `models/` contains functions for converting these abstract sorting options into SQL For instance, `RepoOrderByAlphabetically` is mapped to the SQL fragment `"name ASC"`. This mapping is clearly DB/SQL-related, so it belongs in `models/`.

Conversely, `modules/util/` contains functionality for converting query strings (e.g. `"alphabetically"`) to the corresponding sorting options (`RepoOrderByAlphabetically`). This logic does not have anything to do with the DB; it merely parses/interprets query parameters into the corresponding, abstract sorting option.

As far as I can tell, none of the code currently in `models/` deals with query parameters, and IMO we should maintain this separation of concerns. Let me know if I'm missing something.

Reply:
@lunny Friendly ping. Let me know what you think

Reply:
Functions `ParseRepoOrderByType` and `ToQueryString` are only for type `RepoOrderByType`. Type `RepoOrderByType` is DB-Related, so that why put them on two packages? These two functions are not directly related with DB, but they are related with the type `RepoOrderByType` which are related with DB.

Reply:
I don't mean to be argumentative, but I believe that moving `ParseRepoOrderByType` and `ToQueryString` to `models/` is a bad design decision.

Prior to this PR, there were several identical `switch` statement in various `routers/` handlers that parsed the "sort" query parameter into a `RepoOrderByType`. This PR merely moves this duplicated code to a helper function (in the same package that the original, duplicated code was in).

I'm skeptical of the argument that `ParseRepoOrderByType` and `ToQueryString` are related to `RepoOrderByType`, which is related to the DB, therefore `ParseRepoOrderByType` and `ToQueryString` belong in `models/`. By way of analogy, many endpoints take a "page" query parameter. This "page" parameter eventually gets turned into a `OFFSET` clause, so it as "DB-related" as the "sort" query parameter. Does this mean that the "page" query parameter should be parsed from a string to a int in `models/`? No, we all agree that parsing the string (a representation that is specific to query parameters) into an int (a neutral representation) should happen in `routers/`. Analogously, parsing the "sort" query parameter (a representation that is specific to query parameters) into a `RepoOrderByType` (a neutral representation) should also occur in `routers/`. 

Right now, *all* of the query-parameter-parsing code is in `routers/`. I think that splitting this code between `routers/` and `models/` will make our codebase more confusing.